main.tf

```diff
@@ -239,7 +241,7 @@ resource "aws_iam_openid_connect_provider" "oidc_provider" {
   # Not available on outposts
   count = local.create_oidc_provider ? 1 : 0
 
-  client_id_list  = distinct(compact(concat(["sts.${local.dns_suffix}"], var.openid_connect_audiences)))
+  client_id_list  = distinct(compact(concat(["sts.amazonaws.com"], var.openid_connect_audiences)))
```

Comment:
Curious why this is being changed to a hard-coded value. Does this affect other AWS partitions, like govcloud?

Reply:
No need for it to be dynamic, and no it does not affect other partitions - https://github.com/boto/botocore/blob/7ee6eeed83ea1bd4c527197fffec8bb08b00c5d7/botocore/data/sts/2011-06-15/service-2.json#L6

Reply:
This may be outside the scope of this PR but the only reason this does not have to be dynamic is due to the use of the STS Global Endpoint, which is US-East-1 in disguise. Would it be better to either switch to the Regional Endpoints(Not all are Enabled by Default so this could be a breaking change), or at least allow for users to decide via a feature toggle?